File: server/BulletinBoard.Domain/Models/Advertisement.cs
namespace BulletinBoard.Domain.Models;

public class Advertisement
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public AdvertisementCategory Category { get; set; }
    public decimal? Price { get; set; }
    public Location? Location { get; set; }
    public string CreatedByUserId { get; set; } = string.Empty;
    public DateTimeOffset CreatedAtUtc { get; set; }
    public DateTimeOffset? UpdatedAtUtc { get; set; }
}

File: client/src/app/models/advertisement.ts
import { AdvertisementCategory } from './advertisement-category';
import { Location } from './location';

export interface Advertisement {
    id: string;
    title: string;
    description: string;
    category: AdvertisementCategory;
    price?: number;
    location?: Location;
    createdByUserId: string;
    createdAtUtc: string;
    updatedAtUtc?: string;
    isOwnedByCurrentUser: boolean;
}